app/src/main/java/de/danoeh/antennapod/fragment/QueueFragment.java

```diff
@@ -598,8 +599,11 @@ private void refreshInfoBar() {
         String info = queue.size() + getString(R.string.episodes_suffix);
         if(queue.size() > 0) {
             long timeLeft = 0;
-            float playbackSpeed = UserPreferences.getPlaybackSpeed();
             for(FeedItem item : queue) {
+                float playbackSpeed = item.getFeedPlaybackSpeed();
+                if (playbackSpeed == SPEED_USE_GLOBAL) {
+                    playbackSpeed = UserPreferences.getPlaybackSpeed();
```

Comment:
Video vs audio. Maybe UserPreferences.getPlaybackSpeed should be changed to accept the media type as parameter. Then it is no longer possible to forget the video speed anywhere. Additionally, the name "getPlaybackSpeed" is now a bit misleading. Maybe "getGlobalPlaybackSpeed"? I am not sure how much work this is.

Reply:
Good idea. If it's only a couple of places I can include it.